### PR TITLE
[Feature](pipeline) Trace pipeline scheduling (part I)

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

now we got tracing logs like:
```
8500427614500572113-6789189066394059143|task0(0-1ExchangeSourceOperator)|1|139825136383744|74631528697|74631528720||0
8500427614500572113-6789189066394059143|task0(0-0ScanOperator)|17|139825176491776|74631533384|74631533818||0
8500427614500572113-6789189066394059143|task0(0-0ScanOperator)|3|139825156437760|74631532316|74631533166||0
8500427614500572113-6789189066394059143|task0(0-1ExchangeSourceOperator)|1|139825136383744|74631539128|74631539137||0
8500427614500572113-6789189066394059143|task0(0-0ScanOperator)|17|139825176491776|74631539144|74631539215|RUNNABLE|0
8500427614500572113-6789189066394059143|task0(0-0ScanOperator)|3|139825156437760|74631539137|74631539344|RUNNABLE|0
8500427614500572113-6789189066394059143|task0(0-1ExchangeSourceOperator)|1|139825136383744|74631539341|74631539545|RUNNABLE|0
```

in next pr, we will use a tool to extract-transform it to fit some display tools

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/data_sink.cpp [hunk 1]
```diff
@@ -54,6 +54,24 @@ std::shared_ptr<QueryStatistics> DataSink::get_query_statistics_ptr() {
     return _query_statistics;
 }
 
+static bool _has_inverted_index_or_partial_update(TOlapTableSink sink) {
```
Comment (zclllyybb) [inline be/src/exec/data_sink.cpp:57]: extract to thrift utils

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by Gabriel39 — DISMISSED** [review]

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Comment by zclllyybb** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**